spec/mongo/cursor_spec.rb

```diff
@@ -353,7 +353,7 @@
         cluster.instance_variable_get(:@periodic_executor).flush
         expect do
           cursor.to_a
-        end.to raise_exception(Mongo::Error::OperationFailure, /[cC]ursor.*not found/)
+        end.to raise_exception(Mongo::Error::SessionEnded)
```

Comment:
This exception shouldn't be raised to users, right? If so I suppose this test is OK with a note explaining why this exact exception is raised (cursor cleanup triggers session cleanup, subsequent operation with this session fails).

If this is triggerable by users I would say cursor should track its state and raise a better error prior to attempting an operation on closed session.

Reply:
This exception won't be raised to users. Code that causes is is scheduled in finalizer, so the cursor is gone when the code is executed.